gms/gossiper.cc

```diff
@@ -2309,11 +2312,20 @@ future<> gossiper::wait_for_live_nodes_to_show_up(size_t n) {
     constexpr auto timeout_delay = std::chrono::seconds{30};
 #endif
     auto timeout = gossiper::clk::now() + timeout_delay;
+    auto max_timeout = gossiper::clk::now() + std::chrono::minutes{6};
+    std::unordered_set<inet_address> pending_mark_alive_endpoints;
     while (get_live_members().size() < n) {
-        if (timeout <= gossiper::clk::now()) {
+        auto now = gossiper::clk::now();
+        if (timeout <= now) {
+          if (_pending_mark_alive_endpoints != pending_mark_alive_endpoints && max_timeout > now) {
+            logger.info("pending_mark_alive_endpoints changed while waiting for live nodes: {}", _pending_mark_alive_endpoints);
```

Comment:
Note that if we reach timeout, this is always going to be printed, because `pending_mark_alive_endpoints` is initialized to `nullopt`. So even if `_pending_mark_alive_endpoints` doesn't change during this `timeout_delay`, the message will incorrectly say that it did.

I think we should initialize it to `_pending_mark_alive_endpoints`.

Reply:
It's going to be printed if `_pending_mark_alive_endpoints` is non-null at timeout time.
But I agree it would be more elegant to initialize `pending_mark_alive_endpoints` to `_pending_mark_alive_endpoints` so to detect the first change, if any, more accurately.